### PR TITLE
Validate exact expected error in signed header verification tests

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1049

Stacked on top of #1162

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: types/header.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ type Hash = header.Hash
 // BaseHeader contains the most basic data of a header
 type BaseHeader struct {
 	// Height represents the block height (aka block number) of a given header
-	Height uint64
+	Height int64
```
[the remark this answers is not in the document]
Comment (Manav-Aggarwal) [reply types/header.go]: Sure, will resolve this in a separate PR. Currently, when we generate random uint64 numbers, we attempt to convert them to int64 types in some places, which makes the height negative. Ideally, we just use uint64 everywhere for heights.

File: types/signed_header_test.go [hunk 2]
```diff
@@ -15,89 +16,141 @@ func TestVerify(t *testing.T) {
...
+						case *ErrNewHeaderTimeFromFuture:
+							assert.Equal(t, &ErrNewHeaderTimeFromFuture{}, testReason)
+						default:
+							assert.Equal(t, testReason, reason)
+						}
+					default:
+						assert.Failf(t, "unexpected error: %s\n", err.Error())
+					}
+				}
 			}
```
Comment (MSevey) [inline types/signed_header_test.go:154]: I think we are missing a test case, and we can see that better if we focus on de-indenting to handle case by case.
```suggestion
                 // Case 1 & 2: test.err == nil and err == nil or err != nil
	        if test.err == nil {
		        assert.NoError(t, err)
		        return
	        }
	        // Case 3: test.err != nil and err != nil
	        if err == nil {
	          t.Errorf("expected err: %v, got nil",test.err)
	          return
	        }
		// Case 4: test.err != nil and err != nil
	        switch (err).(type) {
	        case *header.VerifyError:
		        reason := err.(*header.VerifyError).Reason
		        testReason := test.err.(*header.VerifyError).Reason
		        switch reason.(type) {
		        case *ErrLastHeaderHashMismatch:
			        assert.Equal(t, &ErrLastHeaderHashMismatch{}, testReason)
		        case *ErrLastCommitHashMismatch:
			        assert.Equal(t, &ErrLastCommitHashMismatch{}, testReason)
		        case *ErrNewHeaderTimeBeforeOldHeaderTime:
			        assert.Equal(t, &ErrNewHeaderTimeBeforeOldHeaderTime{}, testReason)
		        case *ErrNewHeaderTimeFromFuture:
			        assert.Equal(t, &ErrNewHeaderTimeFromFuture{}, testReason)
		        default:
			        assert.Equal(t, testReason, reason)
		        }
	        default:
		        assert.Failf(t, "unexpected error: %s\n", err.Error())
	        }
```

**Review by MSevey — APPROVED** [review]

**Review by nashqueue — APPROVED** [review]
Thanky you !

File: types/header.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nashqueue) [reply types/header.go]: Follow up: https://github.com/rollkit/rollkit/issues/1176

**Pull request merged**